### PR TITLE
Turn large or mutable structs back into classes

### PR DESCRIPTION
**Pull request by pakrym**
Contributes to https://github.com/Azure/azure-sdk-for-net/issues/6897.

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs [hunk 1]
```diff
@@ -355,66 +355,13 @@ private void EnsureState()
         /// <returns>True if they're equal, false otherwise.</returns>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override bool Equals(object obj)
-            => obj is BlobSasBuilder other && Equals(other);
+            => base.Equals(obj);
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs]: Should we just remove this?

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs]: Oh, nevermind, it's for the [EditorBrowsable(...)]

**Review by tg-msft — APPROVED** [review]
This looks great.  Thank you!

**Comment by JoshLove-msft** [comment]
I think BlobRestClient wasn't regenerated after AccountInfo change. AccountInfo should be a class, correct? I can merge this in my PR, but just wanted to make sure.

**Comment by tg-msft** [comment]
Yes, @JoshLove-msft, there's a separate fix out for that but feel free to merge it if you're going in first.

**Pull request merged**